src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.util.Locale;
+
+/**
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    // protected String by;
+    private LocalDate deadline;
+
+    /**
+     * Constructs a Deadline task.
+     *
+     * @param description   The description of the task.
+     * @param deadlineString  The deadline date string in "yyyy-MM-dd format.
+     */
+    public Deadline(String description, String deadlineString) {
+        super(description);
+        this.deadline = parseDeadline(deadlineString);
+    }
+
+    /**
+     * Parses the deadline date from string.
+     *
+     * @param deadlineString  The deadline date string in "yyyy-MM-dd" format.
+     * @return The parsed deadline date as a LocalDate object, or null if parsing fails.
+     */
+    private LocalDate parseDeadline(String deadlineString) {
+
```

Comment:
I'm not sure if this is an accident but there should be consistency in whether you separate your block statements with 1 line spacing or without.